workers/tasks/export/pandoc.ts

```diff
@@ -186,21 +184,31 @@ export const exportWithPandoc = async (options: ExportWithPandocOptions) => {
 	const metadataFile = await createYamlMetadataFile(pubMetadata, pandocTarget);
 	const bibliographyFile = await createCslJsonBibliographyFile(pandocNotes);
 	const pandocFlags = getPandocFlags(options);
-	const pandocArgs = createPandocArgs(pandocTarget, pandocFlags, metadataFile, bibliographyFile);
+	const pandocArgs = createPandocArgs(
+		pandocTarget,
+		pandocFlags,
+		tmpFile.path,
+		metadataFile,
+		bibliographyFile,
+	);
 	const preTransformedPandocAst = fromProsemirror(pubDoc, rules, {
 		prosemirrorDocWidth: 675,
 		resources: createResources(pandocNotes),
 	}).asNode();
 	const pandocAst = runTransforms(preTransformedPandocAst);
 	const pandocJson = emitPandocJson(pandocAst);
-	const { output, error } = callPandoc(pandocJson, pandocArgs);
+	const error = callPandoc(pandocJson, pandocArgs);
 	if (error) {
 		throw new Error(error);
 	}
-	const transformedOutput = modifyJatsContentToIncludeUnstructuredNotes(
-		output,
-		pandocTarget,
-		pandocNotes,
-	);
-	fs.writeFileSync(tmpFile.path, transformedOutput);
+	// At this point, pandoc has written the document to our temp file. Here
+	// we do some additional post-processing.
+	if (pandocTarget === 'jats_archiving') {
+		const pandocOutput = fs.readFileSync(tmpFile.path).toString();
+		const transformedOutput = modifyJatsContentToIncludeUnstructuredNotes(
+			pandocOutput,
+			pandocNotes,
+		);
+		fs.writeFileSync(tmpFile.path, transformedOutput);
+	}
```

Comment:
I moved the check here because overwriting the contents of docx/epub files was corrupting the contents too (even when it should have been a no-op). Still not totally sure why.